examples/tutorials/PytorchLightning_Integration.ipynb

```diff
@@ -0,0 +1,464 @@
+    "        return torch.utils.data.DataLoader(\n",
+    "            self.train_dataset,\n",
+    "            batch_size=self._batch_size,\n",
+    "            collate_fn=collate_smiles_dataset_wrapper,\n",
+    "            shuffle=True,  \n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4115ec6e",
+   "metadata": {},
+   "source": [
+    "**Create the lightning module**: in this part we create the GCN specific lightning module. This class specifies the logic flow for the training step. We also create the required models, optimizers and losses for the training flow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "61fdd620",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# prepare the LightningModule\n",
+    "class GCNModule(pl.LightningModule):\n",
+    "    def __init__(self, mode, n_tasks, learning_rate):\n",
+    "        super().__init__()\n",
+    "        self.save_hyperparameters(\n",
+    "            \"mode\",\n",
+    "            \"n_tasks\",\n",
+    "            \"learning_rate\",\n",
+    "        )\n",
+    "        self.gcn_model = GCNModel(\n",
+    "            mode=self.hparams.mode,\n",
+    "            n_tasks=self.hparams.n_tasks,\n",
+    "            learning_rate=self.hparams.learning_rate,\n",
+    "        )\n",
+    "        self.pt_model = self.gcn_model.model\n",
+    "        self.loss = self.gcn_model._loss_fn\n",
+    "        \n",
+    "    def configure_optimizers(self):\n",
+    "        return self.gcn_model.optimizer._create_pytorch_optimizer(\n",
+    "            self.pt_model.parameters(),\n",
+    "        )\n",
+    "    \n",
+    "    def training_step(self, batch, batch_idx):\n",
+    "        batch = batch.batch_list\n",
+    "        inputs, labels, weights = self.gcn_model._prepare_batch(batch)\n",
+    "        outputs = self.pt_model(inputs)\n",
+    "        \n",
+    "        if isinstance(outputs, torch.Tensor):\n",
+    "            outputs = [outputs]\n",
+    "    \n",
+    "        if self.gcn_model._loss_outputs is not None:\n",
+    "            outputs = [outputs[i] for i in self.gcn_model._loss_outputs]\n",
+    "    \n",
+    "        loss_outputs = self.loss(outputs, labels, weights)\n",
+    "        \n",
+    "        self.log(\n",
+    "            \"train_loss\",\n",
+    "            loss_outputs,\n",
+    "            on_epoch=True,\n",
+    "            sync_dist=True,\n",
+    "            reduce_fx=\"mean\",\n",
+    "            prog_bar=True,\n",
+    "        )\n",
+    "        \n",
+    "        return loss_outputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c3b167c5",
+   "metadata": {},
+   "source": [
+    "**Create the relevant objects**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "bfff80bd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create module objects\n",
+    "gcn_datasetmodule = GCNDatasetModule(\n",
```

Comment:
Perhaps useful to have a conversion function or example showing how to convert a DeepChem Dataset object into a DatasetModule object.

Reply:
Hi Bharath, it is tough to create a generic function here to convert any deepchem dataset object into a datasetmodule as I think it will need a good amount of wrapper implementation to handle the complexity. If it is okay I would like to keep the current design for the tutorial and we can implement a conversion function in the next PR.